crates/router/src/core/webhooks.rs

```diff
@@ -714,10 +714,39 @@ pub async fn webhooks_core<W: types::OutgoingWebhookType>(
             merchant_account.merchant_id.clone(),
         )],
     );
+    let merchant_connector_account = if connector_name_or_id.starts_with("mca_") {
+        state
+            .store
+            .find_by_merchant_connector_account_merchant_id_merchant_connector_id(
+                &merchant_account.merchant_id,
+                connector_name_or_id,
+                &key_store,
+            )
+            .await
+            .to_not_found_response(errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
+                id: connector_name_or_id.to_string(),
+            })
+            .attach_printable("error while fetching merchant_connector_account from connector_id")?
+    } else {
+        state
+            .store
+            .find_merchant_connector_account_by_merchant_id_connector_label(
+                &merchant_account.merchant_id,
+                connector_name_or_id,
+                &key_store,
+            )
+            .await
+            .to_not_found_response(errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
+                id: connector_name_or_id.to_string(),
+            })
+            .attach_printable("error while fetching merchant_connector_account from connector_id")?
```

Comment:
This will fail if connector is passed, get the connector label and then query
```rust
 let connector_label = utils::get_connector_label_using_object_reference_id(
            db,
            object_reference_id,
            merchant_account,
            connector_name,
        )
        .await
        .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)
        .attach_printable("Error while fetching connector_label")?;
```